api/environments/job_handler.go

```diff
@@ -71,6 +71,35 @@ func (eh EnvironmentHandler) GetJob(appName, envName, jobComponentName, jobName
 	return jh.GetJob(appName, envName, jobComponentName, jobName)
 }
 
+// StopJob Stop job by name
+func (eh EnvironmentHandler) StopJob(appName, envName, jobComponentName, jobName string) error {
+	batchName, batchJobName, ok := parseBatchAndJobNameFromScheduledJobName(jobName)
+	if !ok {
+		return jobNotFoundError(jobName)
+	}
+
+	batch, err := eh.getRadixBatch(appName, envName, jobComponentName, batchName, "")
+	if err != nil {
+		return err
+	}
+
+	idx := slice.FindIndex(batch.Spec.Jobs, func(job radixv1.RadixBatchJob) bool { return job.Name == batchJobName })
+	if idx == -1 {
+		return jobNotFoundError(jobName)
+	}
+
+	jobStatus := slice.FindAll(batch.Status.JobStatuses, func(js radixv1.RadixBatchJobStatus) bool {
+		return js.Name == batchJobName && js.Phase != radixv1.BatchJobPhaseActive && js.Phase != radixv1.BatchJobPhaseWaiting && js.Phase != ""
+	})
+	if len(jobStatus) > 0 {
+		return radixhttp.ValidationError(jobName, fmt.Sprintf("invalid job running state=%s", jobStatus[0].Phase))
+	}
```

Comment:
A job without an entry in JobStatuses is considered to be waiting, and is stoppable. Consider moving the phase filter out of FindAll, and validate the Phase in line 94.

Reply:
the no status is already accounted for, but the validation will be rewritten to be easier to read

Reply:
fixed